server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/DatafileTransactionLog.java

```diff
@@ -0,0 +1,385 @@
+      this.initialFiles = this.finalFiles = files.toArray(new StoredTabletFile[0]);
+      this.updateCount = this.tabletLog.getUpdateCount();
+    }
+
+    /**
+     * Add the passed in transaction, adjusting the log size as needed.
+     *
+     * @param transaction The new transaction
+     */
+    public void addTransaction(DatafileTransaction transaction) {
+      DatafileTransaction removed = this.tabletLog.add(transaction);
+      if (removed != null) {
+        this.initialTs = removed.ts;
+        this.initialFiles = applyTransaction(this.initialFiles, removed);
+      }
+      this.finalFiles = applyTransaction(this.finalFiles, transaction);
+      this.updateCount = this.tabletLog.getUpdateCount();
+    }
+
+    /**
+     * Apply a transaction to the set of files and return an updated file set
+     *
+     * @param files The initial files
+     * @param transaction The transaction
+     * @return The final files
+     */
+    private static StoredTabletFile[] applyTransaction(StoredTabletFile[] files,
+        DatafileTransaction transaction) {
+      Set<StoredTabletFile> newFiles = new HashSet<>(Arrays.asList(files));
+      transaction.apply(newFiles);
+      return newFiles.toArray(new StoredTabletFile[0]);
+    }
+
+    Date getInitialDate() {
+      return Date.from(Instant.ofEpochMilli(initialTs));
+    }
+
+    int getNumTransactions() {
+      return tabletLog.size();
+    }
+
+    List<DatafileTransaction> getTransactions() {
+      return tabletLog.toList();
+    }
+
+    Set<StoredTabletFile> getExpectedFiles() {
+      return new HashSet<>(Arrays.asList(finalFiles));
+    }
+
+    boolean isEmpty() {
+      return tabletLog.isEmpty();
+    }
+
+    /**
+     * Return a human readable dump of the log
+     *
+     * @param extent The tablet extent
+     * @return A dump of the log
+     */
+    String dumpLog(KeyExtent extent, boolean clear) {
+      // first lets get a consistent set of files and transactions
+      long initialTs;
+      StoredTabletFile[] initialFiles;
+      List<DatafileTransaction> transactions;
+      StoredTabletFile[] finalFiles;
+      boolean consistent;
+      do {
+        long updateCount = this.updateCount;
+        initialTs = this.initialTs;
+        initialFiles = this.initialFiles;
+        finalFiles = this.finalFiles;
+        transactions = this.tabletLog.toList();
+        consistent =
+            (updateCount == this.updateCount && updateCount == this.tabletLog.getUpdateCount());
+      } while (!consistent);
+
+      // now clear out the log if requested
+      if (clear) {
+        clear();
+      }
+
+      // now we can build our dump string
+      StringBuilder builder = new StringBuilder();
+      SimpleDateFormat format = new SimpleDateFormat(DATE_FORMAT);
+      String initialDate = format.format(Date.from(Instant.ofEpochMilli(initialTs)));
+
+      builder.append(String.format("%s: Initial files for %s : %s", initialDate, extent,
```

Comment:
Putting the Initial files on its own line would make the output more consistent. 

```suggestion
      builder.append(String.format("\n%s: Initial files for %s : %s", initialDate, extent,
```

Reply:
sure.